### PR TITLE
ISPN-11568 Remove InfinispanIndexManager and dependencies

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11568

### CODE REVIEW

**Comment by anistor** [comment]
:balloon: :tada: :+1:

**Comment by gustavonalle** [comment]
This should be read for reviewing

**Comment by anistor** [comment]
https://www.youtube.com/watch?v=Mx9z99YJ_7s&t=48 :rofl:

**Comment by gustavonalle** [comment]
Regarding documentation, I'll handle it separately, as it probably needs a full review with so many changes on 11

File: core/src/main/java/org/infinispan/configuration/cache/IndexOverlay.java [hunk 1]
```diff
@@ -22,21 +22,6 @@ void apply(TypedProperties properties) {
...
          properties.putIfAbsent(READER_STRATEGY, "shared");
       }
-   },
-
-   /**
-    * Default for indexing a distributed cache. It will store indexes
-    * in Infinispan itself, with a master/slave backend in order to
-    * avoid holding cluster-wide locking during indexing
-    */
-   DISTRIBUTED_INFINISPAN {
```
Comment (anistor) [inline core/src/main/java/org/infinispan/configuration/cache/IndexOverlay.java:32]: How about just deleting the IndexOverlay enum altogether and putting those autoconfigs directly in the builder as private constants ?

**Comment by anistor** [comment]
So are we completely removing the ispn lucene directory @gustavonalle ?

File: core/src/main/java/org/infinispan/configuration/cache/IndexOverlay.java (hunk 1 above)
Comment (gustavonalle) [reply core/src/main/java/org/infinispan/configuration/cache/IndexOverlay.java:32]: sounds good

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/configuration/cache/IndexOverlay.java:32]: done

File: query/src/test/resources/testconfig-LocalCachePerformantConfTest.xml [hunk 2]
```diff
@@ -27,6 +27,7 @@
             </indexed-entities>
             <!-- Enabled fastest writer: NRT backend -->
             <property name="default.indexmanager">near-real-time</property>
+            <property name="default.directory_provider">local-heap</property>
             <property name="default.indexBase">${java.io.tmpdir}/LocalCachePerformantConfTest</property>
```
Comment (anistor) [inline query/src/test/resources/testconfig-LocalCachePerformantConfTest.xml:31]: I believe we do not need default.indexBase for local-heap

File: query/src/test/java/org/infinispan/query/blackbox/ClusteredDistCacheTest.java [hunk 3]
```diff
@@ -18,9 +18,10 @@
    @Override
    protected void createCacheManagers() throws Throwable {
       ConfigurationBuilder cacheCfg = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false);
-      cacheCfg.indexing().autoConfig(true)
+      cacheCfg.indexing()
             .enable()
             .addIndexedEntity(Person.class)
+            .addProperty("default.directory_provider", "local-heap")
             .addProperty("error_handler", "org.infinispan.query.helper.StaticTestingErrorHandler");
```
Comment (anistor) [inline query/src/test/java/org/infinispan/query/blackbox/ClusteredDistCacheTest.java:25]: We have 28 places in our tests where ```addProperty("error_handler", "org.infinispan.query.helper.StaticTestingErrorHandler")``` should be replaced with ```addProperty("error_handler", StaticTestingErrorHandler.class.getName())```

File: query/src/test/resources/nrt-performance-writer-infinispandirectory.xml [hunk 4]
```diff
@@ -28,7 +28,7 @@
             <!-- Enabled fastest writer: NRT backend -->
             <property name="default.indexmanager">near-real-time</property>
             <!-- Write indexes in Infinispan -->
-            <property name="default.directory_provider">infinispan</property>
+            <property name="default.directory_provider">local-heap</property>
             <property name="default.chunk_size">128000</property>
             <property name="default.metadata_cachename">LuceneIndexesMetadataOWR</property>
```
Comment (anistor) [inline query/src/test/resources/nrt-performance-writer-infinispandirectory.xml:33]: These and the corresponding caches are not needed.

**Comment by anistor** [comment]
@gustavonalle I've pushed a WIP commit to your branch, performing the changes I suggested.

Also removed some more internal ispn dir cache confings I found in xmls (metadata, locking, index). Let's see what CI has to say about this.

Another thing I noticed, maybe we should also remove integrationtests/as-lucene-directory module ?

**Comment by anistor** [comment]
Looks like I introduced a new failure. I pushed another wip commit to fix it. Let's see.

**Comment by gustavonalle** [comment]
> maybe we should also remove integrationtests/as-lucene-directory module ?

@anistor Yes, looks like I removed only the test and forgot to remove the module!

**Comment by gustavonalle** [comment]
@anistor It looks like ```org.infinispan.query.distributed.MassIndexingTest.testOverlappingMassIndexers``` has a race, it is probably on ```master``` as well, and this PR that changes to ```local-heap``` everywhere may have made it more evident

**Comment by gustavonalle** [comment]
So feel free to disable it for now and I will create a JIRA to fix it properly

**Comment by anistor** [comment]
Last CI run was ok. There are some failures that are also present on master.

I've added another commit to remove integrationtests/as-lucene-directory and will let CI do it's thing again, then I will squash all commits and merge.

**Comment by anistor** [comment]
Merged in master ! Thanks @gustavonalle

**Pull request closed without merging**